### ISSUE TITLE
[3.3.5][Rogue]Killing spree vs Dragon bosses

### ISSUE THREAD

**Issue by Excite** [TrinityCore/TrinityCore#8758]
Still not working part:

3) Killing spree: should be in Sindragosa stomach but he attacks Sindragosa tail and throws me far away.

Fixed part:
<details>

I have encountered a problem with Chilled to the bone debuff + Cloak of shadows + killing spree at Sindragosa 

How to reproduce the bug:
Me as Rogue Combat / Mutilate Cannot do any dps on Sindragosa Why ?

1) The bug with Chilled th the bone is : one Mutilate / one Sinister strike can stack me to 6 stacks and Two times of each skill can stack me to 8-12 stacks.
2)Cloak of Shadows: Should remove the Chill to the bone Stacks and he doesn't removes it

Video that shows every thing i write there: http://www.youtube.com/watch?v=qM3Q7BXZCng 
as you see in the video that the rogue Use Killing spree into the Sindragosa stomach and use cloak of shadows to remove his Chilled to the bone stacks and he can attack like 20 sec without getting more then 5 stacks (i get 6 in 1 sec)

Blood queen -  Swarming Shadows

How to reproduce the bug : 
as you can see in this video 1:50 https://www.youtube.com/watch?v=6rPoaHBzXSY 
The rogue use cloak of shadows and removes the swarming shadows .
Its not happens to us, even not absorb the swarming shadows with cloak of shadows on.

http://www.zeroinitiate.com/the-icecrown-citadel-blood-queen-lanathel/
Quote from the website "The final debuff is Swarming Shadows. It’s like Jaraxxus’ fire debuff except it’s purple. You can use Cloak of Shadows to remove it even before it drops a trail"

</details>

Hash: 13c8869
Tdb: 335.11.49+
##

Edit -- Keader
Issue confirmed on rev c1b833a919fb9f3fd01bcbb2c23c32cb6e57115f too.

**Comment by Keader** [comment]
Shit...
Killing spree part is not fixed.

**Comment by Keader** [comment]
@chaodhib any idea here?
i think it's happening with all dragon bosses.

**Comment by chaodhib** [comment]
I'm not sure I would be the best person to help. What is the issue exactly?

**Comment by Keader** [comment]
when you use killing spree you should be teleported to dragon stomach (middle of boss) instead behind him

**Comment by chaodhib** [comment]
From SpellWork:
```
57840 Killing Spree
Effect 0: Id 5 (SPELL_EFFECT_TELEPORT_UNITS)
BasePoints = 1
Targets (1, 65) (TARGET_UNIT_CASTER, TARGET_DEST_TARGET_BACK)
Radius (Id 7) 2,00
```

So the attacker should teleport in the target's back.

From sniff in Arathi Bassin (map id 529):
> ServerToClient: MSG_MOVE_ROOT (0x00EC) Length: 37 ConnIdx: 0 Time: 07/16/2010 20:30:20.000 Number: 102737
> Guid: Full: 0x28000000237F2DD Type: Player Low: 37221085 Name: Holystrikerr
> Movement Flags: Root (2048)
> Extra Movement Flags: None (0)
> Time: 3706933533
> Position: X: 852.9475 Y: 1166.165 Z: 10.47833
> Orientation: 5.828441
> Fall Time: 816
> 
> ...
> 
> ServerToClient: SMSG_SPELL_GO (0x0132) Length: 62 ConnIdx: 0 Time: 07/16/2010 20:30:23.000 Number: 103323
> Caster GUID: Full: 0x280000003AAFB17 Type: Player Low: 61537047 Name: Babycake
> Caster Unit GUID: Full: 0x280000003AAFB17 Type: Player Low: 61537047 Name: Babycake
> Cast Count: 0
> Spell ID: Killing Spree (57840)
> Cast Flags: PendingCast, Unknown3, Unknown7 (265)
> Time: 3706936424
> Hit Count: 1
> [0] Hit GUID: Full: 0x280000003AAFB17 Type: Player Low: 61537047 Name: Babycake
> Miss Count: 0
> Target Flags: Unit, DestinationLocation (66)
> Target GUID: Full: 0x28000000237F2DD Type: Player Low: 37221085 Name: Holystrikerr
> Destination Transport GUID: 0x0
> Destination Position: X: 851.1508 Y: 1167.043 Z: 10.44661
> Unk Byte 2: 6

So it's consistent with the spell data: the attacker does indeed teleport behind the target. And it's consistent with what the TC already does. 

The only remaining issue is the distance. In the example above, the caster teleports exactly 2 meters away from his target. In the core, the distance depends on the target's size: the bigger the target, the further away the caster is teleported. Investigation should be done (through sniff and maybe test on live retail) to see if the teleportation distance should be constant (2 meters) or should depend on some factors.

**Comment by chaodhib** [comment]
Oh crap. I went to fast. The distance of teleportation can simply be found in the spell data. We can see that for Killing Spree (spell id 57840), the distance is 2m. Same for Shadowstep. So the issue is simple to fix now.

**Comment by Keader** [comment]
yay =)

**Comment by offl** [comment]
Fixed in 9dbef4b6bb7c9f28e9140782d938f41ed04df6f2?
